### ISSUE TITLE
Ctrl+Hover causes diagnostics request for the definition module

### ISSUE THREAD

**Issue by nadako** [vshaxe/vshaxe#366]
Subj. Probably because of the preview?

**Comment by nadako** [comment]
Blocked by https://github.com/microsoft/vscode/issues/78453, I guess...

**Comment by nadako** [comment]
[Hmmm](https://github.com/microsoft/vscode/issues/78453#issuecomment-518690945)

> That's the designed behaviour because there is only one way of opening a document. The underlying problems seems to be that diagnostics are driven by open/closed documents, not by visible editors nor by visible tabs (the latter has no API)

**Comment by Gama11** [comment]
Sounds like "wait for https://github.com/microsoft/vscode/issues/15178".

**Comment by nadako** [comment]
Or maybe it makes sense to only request diagnostics for the current active text editor... Gotta think/experiment with this.

**Comment by Gama11** [comment]
One problem with that would be that we shouldn't even know about the active editor.. i.e. that's not part of the LSP, but a custom notification method. So by only relying on that, we'd break diagnostics for non-VSCode-LSP clients.

**Comment by nadako** [comment]
Well I guess we could make it configurable so it works in non-VSCode, but it all looks like a workaround... Still I think we should move towards fixing/avoiding this issue as it can cause serious slowdowns.

**Comment by nadako** [comment]
This was brought up again today at our sync meeting at @innogames, because this also slows down "go to definition" with Ctrl/Cmd+Click, since it has to wait for diagnostics. While this surely can be worked around basically by learning to use hotkeys for "go to definition", it would still be nice to fix/workaround in vshaxe.

**Comment by nadako** [comment]
Oh well, I worked this around in https://github.com/vshaxe/haxe-language-server/pull/50
